docs/src/arrays.md

```diff
@@ -0,0 +1,219 @@
+# Arrays
+
+When the inputs and outputs of the functions are arrays (potentially with scalars), a modified version of Giles' method is a quick way of deriving `frule`s and `rrule`s.
+Giles' method is succinctly explained in [this paper](https://people.maths.ox.ac.uk/gilesm/files/NA-08-01.pdf), but we will slightly generalize it.
```

Comment:
Perhaps comment on why it's been generalised, otherwise the reader won't discover it until they read your `!!!note` further down.

Reply:
Good idea, I'll introduce the comparison of the note at the top.